src/utils.c

```diff
@@ -78,15 +80,26 @@ void utils_open_browser(const gchar *uri)
 	g_return_if_fail(uri != NULL);
 	win32_open_browser(uri);
 #else
-	gchar *argv[2] = { (gchar *) uri, NULL };
+	gchar *new_cmd, *argv[2] = { (gchar *) uri, NULL };
 
 	g_return_if_fail(uri != NULL);
 
-	while (!spawn_async(NULL, tool_prefs.browser_cmd, argv, NULL, NULL, NULL))
+	while (1)
 	{
-		gchar *new_cmd = dialogs_show_input(_("Select Browser"), GTK_WINDOW(main_widgets.window),
-			_("Failed to spawn the configured browser command. "
-			  "Please correct it or enter another one."),
+		/* Uses the user's default browser akin to xdg-open (in flatpak through a portal) */
+		if (EMPTY(tool_prefs.browser_cmd))
+		{
+			if (gtk_show_uri_on_window(GTK_WINDOW(main_widgets.window), uri, GDK_CURRENT_TIME, NULL))
```

Comment:
Just be aware of one thing: `gtk_show_uri*()` will happily open a *lot* more URIs than what the web browser might support.  It basically can open *any* URI, be it `http:`, `mailto:` or `my-custom-scheme:`.  It's nice and all, but some suggest that it should be handled with care not to open random programs using random schemes.

Admittedly, there's two things going for us here:
* we're not a chat app or such, we don't present mangled untrusted URIs
* all internal URIs are safe and likely (?) to open in the browser (`https:`, `file:` -- need to check that last one though)

But we cannot know what plugins might do.

Though we should check how are `file:` URIs handled. We use that in help, and when using the builtin build exec command.

Reply:
> we're not a chat app or such, we don't present mangled untrusted URIs

Its __worse than that__!!!! If the build command is "builtin" `utils_open_browser()` is used to open stuff __I__ wrote :grin:

Reply:
Should I change the patch or was this just a "beware!" remark?

Reply:
That was just a "beware!" remark, but it'd be nice to check what happens with `file://` URIs we might have to local help

Reply:
Seems we already feed file:// URIs to the function, e.g. for ` file:///usr/local/share/doc/geany/html/index.html`. So at least `file://` must be supported, although without the change we explicitly opened the configured browser, and now we rely on GTK do find the right "browser".

I hesitate to limit the inputs the function accepts. Webdevelopment plugins may want this function to open arbitrary stuff in the browser.

If anything, we might limit to `https://`, `http://` and `file://` but then we might exclude fancy webapps that some plugins could use (they could use `gtk_show_uri_on_window()` directly but lets help them to work on windows as well).

And then,  on Windows we also do a generic `open`, the `browser_cmd` configuration is not a thing there.

Long story shot: I leave this as-is until problems arise